amlb/utils/core.py

```diff
@@ -234,8 +234,9 @@ def __json__(self):
         return Namespace.dict(self)
 
 
-def repr_def(obj):
-    return "{clazz}({attributes})".format(clazz=type(obj).__name__, attributes=', '.join(("{}={!r}".format(k, v) for k, v in obj.__dict__.items())))
+def repr_def(obj, show_private=False):
+    return "{cls}({attrs!r})".format(cls=type(obj).__name__,
+                                     attrs={k: v for k, v in vars(obj).items() if show_private or not k.startswith('_')})
```

Comment:
Not asking you to change it now, but for readability of the code and (in this case in particular) the diffs, using multiple lines of code is nicer.